src/api/utils.ts

```diff
@@ -79,10 +79,15 @@ export const apiFactory =
         parsedResponse = JSON.parse(textResult);
       } catch (e) {
         // We use the text response if its not an empt string
-        if (textResult !== "") parsedResponse = textResult;
+        if (textResult !== "") parsedResponse = { body: textResult };
       }
       try {
-        return schema.parse(parsedResponse);
+        // if response has content, add headers, otherwise return null
+        const headers = Object.fromEntries(res.headers);
+        if (parsedResponse) {
+          return schema.parse({ ...parsedResponse, headers });
```

Comment:
I wonder if we should not merge the headers into the JSON response and rather divide it into body and headers here as well. 
pros:
- a JSON response would be consistent with the text response
- a JSON response can not have its own `headers` property atm, because it would be overwritten by the headers of the response

cons: 
- we would have to update every response schema to have body and headers

Reply:
solved in https://github.com/direktiv/direktiv-ui/commit/435e4b2f04f15fbfb612773753b5c1131e27b94b